### PR TITLE
More consistent stores

### PR DESCRIPTION
**Pull request by Aylr**
This was a fun PR. One single line to change the behavior and tons of tests to update.

## What's Here

* the expectations_store config section is now nested inside stores
* This obviously required many many fixtures to be updated.
* Improved documentation in the yml about stores
* Cleaned up a few dead imports in a few random files

### CODE REVIEW

File: great_expectations/data_context/data_context.py [hunk 1]
```diff
@@ -212,7 +209,7 @@ def __init__(self, project_config, context_root_dir, data_asset_name_delimiter='
         self._stores = DotDict()
         self.add_store(
             "expectations_store",
-            copy.deepcopy(self._project_config["expectations_store"]),
+            copy.deepcopy(self._project_config["stores"][self._project_config["expectations_store_name"]]),
```
Comment (Aylr) [inline great_expectations/data_context/data_context.py:212]: The actual code change.

File: great_expectations/data_context/data_context.py [hunk 2]
```diff
@@ -187,11 +187,8 @@ def __init__(self, project_config, context_root_dir, data_asset_name_delimiter='
         Returns:
             None
         """
-        # if not isinstance(project_config, DataContextConfig):
-
         if not ConfigOnlyDataContext.validate_config(project_config):
-            raise TypeError("project_config is not valid. Try using the CLI check-config command.")
-
+            raise ge_exceptions.InvalidConfigError("Your project_config is not valid. Try using the CLI check-config command.")
```
Comment (Aylr) [inline great_expectations/data_context/data_context.py:191]: More precise error.

File: great_expectations/data_context/data_context.py [hunk 3]
```diff
@@ -1825,9 +1822,6 @@ def _save_project_config(self):
                 self._project_config
             )
 
-            #the expectation_store shouldn't appear in the list
-            del config["stores"]["expectations_store"]
-
```
Comment (Aylr) [inline great_expectations/data_context/data_context.py:1830]: Bye bye!

File: great_expectations/data_context/templates.py [hunk 4]
```diff
@@ -47,18 +47,21 @@
...
 evaluation_parameter_store_name: evaluation_parameter_store
 
-# Stores are where things like expectations, evalutation results, etc are stored.
 stores:
+  expectations_store:
+    # This is where expectations are kept.
+    class_name: ExpectationStore
+    store_backend:
+      class_name: FixedLengthTupleFilesystemStoreBackend
+      base_directory: expectations/
```
Comment (Aylr) [inline great_expectations/data_context/templates.py]: Better inline docs

File: great_expectations/data_context/templates.py [hunk 5]
```diff
@@ -47,18 +47,21 @@
...
 plugins_directory: plugins/
 
-expectations_store:
-  # This is where expectations are kept.
-  class_name: ExpectationStore
-  store_backend:
-    class_name: FixedLengthTupleFilesystemStoreBackend
-    base_directory: expectations/
-
+# Stores are configurable places to store things like expectations, evaluation
```
Comment (jcampbell) [inline great_expectations/data_context/templates.py:50]: Add:
"Three stores are required: expectations, profiling, and evaluation parameters, and must be named here. Additional stores can be configured below and used for data_docs, validation_operators or other purposes"

**Review by jcampbell — CHANGES_REQUESTED** [review]
<3 One small tweak to the wording.

File: great_expectations/data_context/templates.py (hunk 5 above)
Comment (Aylr) [reply great_expectations/data_context/templates.py:50]: Love it.

**Review by jcampbell — APPROVED** [review]

**Pull request merged**